src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -402,10 +402,15 @@ public override Symbol VisitAssembly(AssemblySymbol symbol)
                     return _otherAssembly;
                 }
 
-                // find a referenced assembly with the exactly same identity:
+                // find a referenced assembly with the exactly same source identity:
                 foreach (var otherReferencedAssembly in _otherAssembly.Modules[0].ReferencedAssemblySymbols)
                 {
-                    if (symbol.Identity.Equals(otherReferencedAssembly.Identity))
+                    var identity = symbol.Identity;
+                    var otherIdentity = otherReferencedAssembly.Identity;
+
+                    if (AssemblyIdentityComparer.SimpleNameComparer.Equals(identity.Name, otherIdentity.Name) &&
+                        symbol.AssemblyVersionInSource.Equals(otherReferencedAssembly.AssemblyVersionInSource) &&
```

Comment:
Do we need to worry about assemblies built by previous version of the compilers?